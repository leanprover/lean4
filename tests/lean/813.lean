open Function
theorem constApply {y : β} {x : α} : const α y x = y := rfl
example (p : Nat → Prop) (h : p 1) : p (const Nat 1 2) := by simp only [constApply]; trace_state; exact h -- simplifies
example (p : α → Prop) (h : p a) : p (const Nat a 2) := by simp only [constApply]; trace_state; exact h -- simplifies
example (f : α → α) (p : α → Prop) (h : p (f a)) : p (const Nat (f a) 2) := by simp only [constApply]; trace_state; exact h -- simplifies
