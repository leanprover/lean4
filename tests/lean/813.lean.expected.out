p : Nat → Prop
h : p 1
⊢ p 1
α : Sort ?u
a : α
p : α → Prop
h : p a
⊢ p a
α : Sort ?u
a : α
f : α → α
p : α → Prop
h : p (f a)
⊢ p (f a)
